gdal/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp

```diff
@@ -1083,12 +1094,14 @@ OGRMySQLDataSource::ICreateLayer( const char * pszLayerNameIn,
 
 /* -------------------------------------------------------------------- */
 /*      Try to get the SRS Id of this spatial reference system,         */
-/*      adding tot the srs table if needed.                             */
+/*      adding to the srs table if needed.                             */
 /* -------------------------------------------------------------------- */
-    int nSRSId = GetUnknownSRID();
-
-    if( poSRS != nullptr )
-        nSRSId = FetchSRSId( poSRS );
+    if( GetMajorVersion() < 8 || IsMariaDB() )
```

Comment:
why is nSRSID fetched conditionnaly to the mysql version ? It seems it is used unconditionnaly later at https://github.com/OSGeo/gdal/pull/2111/files#diff-b546c2f165c017991b14e08089e3d95aR1138 ? So this should probably be moved around line https://github.com/OSGeo/gdal/pull/2111/files#diff-b546c2f165c017991b14e08089e3d95aR1035

Reply:
It seems that we can remove the line https://github.com/OSGeo/gdal/pull/2111/files#diff-b546c2f165c017991b14e08089e3d95aR1101 ' nSRSId = GetUnknownSRID();' because it already set previously at  https://github.com/OSGeo/gdal/pull/2111/files#diff-b546c2f165c017991b14e08089e3d95aR1035 
This is unnecessary call.

Reply:
I still find the fact that nSRSId = FetchSRSId( poSRS ) is called both at line 1048 and 1102 confusing. Cannot just it be done once just after the initialization of line 1035 ?

Reply:
I'd like to move a following blocks on just after line 1048 as such;
```
    int nSRSId = GetUnknownSRID();

    // Calling this does no harm
    InitializeMetadataTables();

/* -------------------------------------------------------------------- */
/*      Try to get the SRS Id of this spatial reference system,         */
/*      adding to the srs table if needed.                             */
/* -------------------------------------------------------------------- */

    if (poSRS != nullptr)
        nSRSId = FetchSRSId(poSRS);
```

This looks works.